### PR TITLE
Fixes for error messages in local exec mode

### PR DESCRIPTION
**Pull request by dankwart-de**

### CODE REVIEW

File: RoddyCore/src/de/dkfz/roddy/execution/io/ExecutionService.groovy [hunk 1]
```diff
@@ -430,15 +430,15 @@ public abstract class ExecutionService extends CacheProvider implements de.dkfz.
      * Checks all the files created in writeFilesForExecution
      * If strict mode is enabled, all missing files will lead to false.
      * If strict mode is disabled, only neccessary files are checked.
-     * @return
+     * @return A descriptive list of missing files
      */
-    boolean checkFilesPreparedForExecution(ExecutionContext context) {
-        if (!context.getExecutionContextLevel().isOrWasAllowedToSubmitJobs) return true
+    List<String> checkFilesPreparedForExecution(ExecutionContext context) {
```
Comment (vinjana) [inline RoddyCore/src/de/dkfz/roddy/execution/io/ExecutionService.groovy]: Would listInaccessibleFiles() or similar not be a better name then?

File: RoddyCore/src/de/dkfz/roddy/execution/io/ExecutionService.groovy [hunk 2]
```diff
@@ -430,15 +430,15 @@ public abstract class ExecutionService extends CacheProvider implements de.dkfz.
...
+        if (!context.getExecutionContextLevel().isOrWasAllowedToSubmitJobs) return []
         boolean strict = Roddy.isStrictModeEnabled()
         def runtimeService = context.getRuntimeService()
 
-        boolean neccessaryFilesExist = true
-        boolean ignorableFilesExist = true
+        List<String> neccessaryFilesExist = []
```
Comment (vinjana) [inline RoddyCore/src/de/dkfz/roddy/execution/io/ExecutionService.groovy]: Would inaccessibleNecessaryFiles not be a better name?

File: RoddyCore/src/de/dkfz/roddy/execution/io/ExecutionService.groovy [hunk 3]
```diff
@@ -430,15 +430,15 @@ public abstract class ExecutionService extends CacheProvider implements de.dkfz.
...
+        List<String> ignorableFilesExist = []
```
Comment (vinjana) [inline RoddyCore/src/de/dkfz/roddy/execution/io/ExecutionService.groovy]: inaccessibleIgnorableFiles?

File: RoddyCore/src/de/dkfz/roddy/execution/io/fs/FileSystemAccessProvider.groovy [hunk 4]
```diff
@@ -335,24 +335,25 @@ public class FileSystemAccessProvider extends CacheProvider {
...
                 String cmd = commandSet.getCheckDirectoryCommand(f, true, outputFileGroup, outputAccessRightsForDirectories);
                 ExecutionResult er = ExecutionService.getInstance().execute(cmd);
-                _directoryExistsAndIsAccessible[path] = (er.firstLine == commandSet.getReadabilityTestPositiveResult());
+                checkGood = (er.firstLine == commandSet.getReadabilityTestPositiveResult());
                 fireCacheValueAddedEvent(id, path);
             } else {
                 String cmd = commandSet.getCheckDirectoryCommand(f)
                 ExecutionResult er = ExecutionService.getInstance().execute(cmd);
-                _directoryExistsAndIsAccessible[path] = (er.firstLine == commandSet.getReadabilityTestPositiveResult());
+                checkGood = (er.firstLine == commandSet.getReadabilityTestPositiveResult());
```
Comment (vinjana) [inline RoddyCore/src/de/dkfz/roddy/execution/io/fs/FileSystemAccessProvider.groovy]: getReadabilityTestPositiveResult? Why not just isReadable?

File: RoddyCore/src/de/dkfz/roddy/execution/io/fs/FileSystemAccessProvider.groovy [hunk 5]
```diff
@@ -335,24 +335,25 @@ public class FileSystemAccessProvider extends CacheProvider {
      */
     public boolean checkDirectory(File f, ExecutionContext context, boolean createMissing) {
         final String path = f.absolutePath
+        boolean checkGood = false
```
Comment (vinjana) [inline RoddyCore/src/de/dkfz/roddy/execution/io/fs/FileSystemAccessProvider.groovy]: directoryIsAccessible?

File: RoddyCore/src/de/dkfz/roddy/execution/io/ExecutionService.groovy (hunk 1 above)
Comment (dankwart-de) [inline RoddyCore/src/de/dkfz/roddy/execution/io/ExecutionService.groovy]: checkForInaccessiblePreparedFiles

File: RoddyCore/src/de/dkfz/roddy/execution/io/ExecutionService.groovy (hunk 2 above)
Comment (dankwart-de) [inline RoddyCore/src/de/dkfz/roddy/execution/io/ExecutionService.groovy]: inaccessibleNecessaryFiles
inaccessibleIgnorableFiles

File: RoddyCore/src/de/dkfz/roddy/execution/io/ExecutionService.groovy (hunk 3 above)
Comment (dankwart-de) [inline RoddyCore/src/de/dkfz/roddy/execution/io/ExecutionService.groovy]: see above

File: RoddyCore/src/de/dkfz/roddy/execution/io/fs/FileSystemAccessProvider.groovy (hunk 4 above)
Comment (dankwart-de) [inline RoddyCore/src/de/dkfz/roddy/execution/io/fs/FileSystemAccessProvider.groovy]: directoryIsAccessible

**Pull request merged**